### PR TITLE
#5876 - retroactive pull request for cba9d16

### PR DESCRIPTION
**Pull request by rseanhall**
do not merge - for comments only https://github.com/wixtoolset/issues/issues/5876

### CODE REVIEW

File: src/wixext/VSCompiler.cs [hunk 1]
```diff
@@ -73,186 +60,171 @@ public override void ParseElement(XElement parentElement, XElement element, IDic
...
 
-            if (!this.Core.EncounteredError)
+            if (!this.Messaging.EncounteredError)
             {
-                Row row = this.Core.CreateRow(sourceLineNumbers, "HelpNamespace");
-                row[0] = id;
-                row[1] = name;
-                row[2] = fileId;
-                row[3] = description;
+                var row = this.ParseHelper.CreateRow(section, sourceLineNumbers, "HelpNamespace", id);
```
Comment (rseanhall) [inline src/wixext/VSCompiler.cs]: I was surprised to see this is called CreateRow when it's actually creating an IntermediateTuple. I'd rather not leave it like that.

File: src/wixext/VSCompiler.cs [hunk 2]
```diff
@@ -73,186 +60,171 @@ public override void ParseElement(XElement parentElement, XElement element, IDic
...
+                row.Set(1, name);
+                row.Set(2, fileId);
+                row.Set(3, description);
```
Comment (rseanhall) [inline src/wixext/VSCompiler.cs]: If we're not going to always use the named property, why doesn't the tuple support the row[1] = name syntax?

**Comment by barnson** [comment]
Even better than an issue!

File: src/wixext/VSCompiler.cs (hunk 1 above)
Comment (rseanhall) [reply src/wixext/VSCompiler.cs:212]: Fixed in https://github.com/wixtoolset/Extensibility/pull/9

File: src/wixext/VSCompiler.cs (hunk 2 above)
Comment (rseanhall) [reply src/wixext/VSCompiler.cs:215]: We should always be using the named property with a strongly typed tuple.

**Pull request closed without merging**